src/components/ProfilePicPopup.js

```diff
@@ -0,0 +1,83 @@
+import Popup from "./Popup";
+
+class ProfilePicPopup extends Popup {
+  constructor({ popupSelector, avatarSelector, handleFormSubmit }) {
+    super(popupSelector);
+    this._popupForm = this.popupElement.querySelector(".popup__form");
+    this._submitButton = this._popupForm.querySelector(".popup__button");
+    this._defaultButtonText = this._submitButton.textContent;
+    this._avatarElement = document.querySelector(avatarSelector);
+    this._handleFormSubmit = handleFormSubmit;
+    this._inputList = Array.from(
+      this._popupForm.querySelectorAll(".popup__input")
+    );
+    this._setEventListeners();
+  }
+
+  _getInputValues() {
+    const userInput = this._inputList.reduce((acc, input) => {
+      const { name, value } = input;
+      acc[name] = value;
+      return acc;
+    }, {});
+    return userInput;
+  }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = this._defaultButtonText;
+    }
```

Comment:
**COULD BE IMPROVED**

`"Saving..."` is hardcoded here

It’s better to make a `renderLoading` method inside `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```

Reply:
Thanks for the idea with the 2nd argument, I did not think about it. 
Done.